### PR TITLE
feat: add doc about compatible plugins...

### PR DESCRIPTION
**Pull request by nickboldt**
### What does this PR do?

feat: add doc about compatible plugins (RHIDP-4150)

Signed-off-by: rhdh-bot service account <rhdh-bot@redhat.com>

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?
[RHIDP-4150](https://issues.redhat.com/browse/RHIDP-4150)

### How to test this PR?
See html preview at link below

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works, if one exists](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections What issues does this PR fix or reference and How to test this PR completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: modules/release-notes/con-relnotes-notable-features.adoc [hunk 1]
```diff
@@ -61,4 +61,8 @@ For more information, see the link:{LinkAdminGuide}#proc-viewing-installed-plugi
 
 == Supported plugins in {product} {product-version}
 
-To get a comprehensive list of supported dynamic plugins, see the link:{LinkAdminGuide}#rhdh-supported-plugins[Dynamic plugins included in {product}] section in the _Administration guide_.
+To get a comprehensive list of supported dynamic plugins, see the link:{LinkPluginsGuide}#rhdh-supported-plugins[Dynamic plugins included in {product}] section in the _{NameOfPluginsGuide} guide_.
+
+== Compatible plugins in {product} {product-version}
+
+To get a comprehensive list of compatible dynamic plugins, see the link:{LinkPluginsGuide}#rhdh-compatible-plugins[Compatible plugins] section in the _{NameOfPluginsGuide} guide_.
```
Comment (Preeticp) [inline modules/release-notes/con-relnotes-notable-features.adoc:68]: Suggestion: Replace lines 64 to 68 with: 

The documentation now lists multiple groups of supported plugins. See these sections in the documentation for further details:

- Red Hat Generally Available plugins
- Red Hat Technology Preview plugins
- Community plugins
- Other compatible plugins
These bulleted items could be hyperlinks to the documentation sections. (@Gerry-Forde might fine tune this when the restructured plugin docs are in place.)

Comment (nickboldt) [reply modules/release-notes/con-relnotes-notable-features.adoc:68]: restructured plugin docs are already in place. I'll updatemy PR with the links.

**Comment by nickboldt** [comment]
Previews:

RN: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-525/rel-notes-rhdh/#_available-plugins-in-red-hat-developer-hub-1-3
Supported: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-525/plugin-rhdh/#rhdh-supported-plugins
TP: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-525/plugin-rhdh/#rhdh-tech-preview-plugins
Community: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-525/plugin-rhdh/index#rhdh-community-plugins
Other compatible: https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-525/plugin-rhdh/#rhdh-compatible-plugins

File: assemblies/dynamic-plugins/assembly-rhdh-installing-dynamic-plugins.adoc [hunk 2]
```diff
@@ -33,12 +33,24 @@ include::../modules/dynamic-plugins/ref-rh-supported-plugins.adoc[leveloffset=+3
...
 
-// Red Hat technology preview plugins
+// Red Hat Technology Preview plugins
+[id="rhdh-tech-preview-plugins"]
 include::../modules/dynamic-plugins/ref-rh-tech-preview-plugins.adoc[leveloffset=+4]
 
+[NOTE]
+====
+* A new Technology Preview plugin for Red Hat Ansible Automation Platform (RHAAP) is available, which replaces this older one. See link:{LinkPluginsGuide}#rhdh-compatible-plugins[Other compatible plugins] in the _{NameOfPluginsGuide} guide_ for further details.
+====
```
Comment (themr0c) [inline assemblies/dynamic-plugins/assembly-rhdh-installing-dynamic-plugins.adoc:43]: We might want to keep this kind of information (previously ... currrently) only in release notes (for maintenance purpose).

File: modules/release-notes/con-relnotes-notable-features.adoc (hunk 1 above)
Comment (Preeticp) [reply modules/release-notes/con-relnotes-notable-features.adoc]: > Nick, Re. restructured plugin docs are already in place. 
@Gerry-Forde has a draft for the restructure, which could slightly tweak the changes in this PR. I will try to point out some in this review as well. We will share the draft with all of you before we merge it though.

**Comment by Preeticp** [comment]
WRT:
image
JSYK, we plan to remove the 2.3 Title (Preinstalled dynamic plugin descriptions and details) and make the tables part of the 2.2 Assembly as part of the restructure.
Let's remove the numbers, 17, 7 etc. It adds to the maintenance with support level changes that depend on release cycles apart from RHDH and the number of plugins we support is also indicated through the length of our table.

File: modules/release-notes/con-relnotes-notable-features.adoc (hunk 1 above)
Comment (Gerry-Forde) [reply modules/release-notes/con-relnotes-notable-features.adoc]: Draft restructure doc: https://docs.google.com/document/d/1YdQJyXc4cJrsDNvk3yjDQMvfI0gLzv6J9AGRXbQgEP4

Comment (Gerry-Forde) [reply modules/release-notes/con-relnotes-notable-features.adoc]: Full restructure doc is available [here](https://docs.google.com/document/d/1YdQJyXc4cJrsDNvk3yjDQMvfI0gLzv6J9AGRXbQgEP4/edit?usp=sharing).

File: assemblies/dynamic-plugins/assembly-rhdh-installing-dynamic-plugins.adoc (hunk 2 above)
Comment (nickboldt) [reply assemblies/dynamic-plugins/assembly-rhdh-installing-dynamic-plugins.adoc:43]: well, Ben's idea was to call out that there's a new plugin that will replace the old one, in situ with the list of plugins.

**Comment by nickboldt** [comment]
> Let's remove the numbers, 17, 7 etc. please. It adds to the maintenance
Numbers are generated by the script but I can remove them if you feel they don't add value. Maintenance is the same as for the plugin names and versions: run the script to get the latest information.

**Comment by Gerry-Forde** [comment]
@Preeticp AFAIU the "compatible" plugins have the same support status as pre-installed "community" plugins insofar as Red Hat carries out integration testing to validate basic functionality of these plugins. 

If this is the case, we can call that out (as we do for community plugins), but we currently link to the Developer Hub Support Policy to clarify this for community plugins, so we may need to update that page for "compatible" plugins ... or maybe just handle it in the product docs with a note above the 'compatible" plugins table.

I'd like to get @reestr view on the support status for "compatible" plugins and how we should document this.

**Comment by nickboldt** [comment]
> AFAIU the "compatible" plugins have the same support status as pre-installed "community" plugins

Yes and all three (RH TP, RH compatible, Community) fall under Technology Preview support statements.

**Comment by Preeticp** [comment]
> @Preeticp AFAIU the "compatible" plugins have the same support status as pre-installed "community" plugins insofar as Red Hat carries out integration testing to validate basic functionality of these plugins.
> 
> If this is the case, we can call that out (as we do for community plugins), but we currently link to the Developer Hub Support Policy to clarify this for community plugins, so we may need to update that page for "compatible" plugins ... or maybe just handle it in the product docs with a note above the 'compatible" plugins table.
> 
> I'd like to get @reestr view on the support status for "compatible" plugins and how we should document this.

+1 i would like @reestr view too.
AFAIU from our conversation on the google doc, the compatible plugins could go GA at a future date. But that is beside my point of asking RHDH to be replaced by Red Hat here:[ "might want to mention RH instead of RHDH provides Technology Preview support for the following compatible plugins"](https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/pull/525#issuecomment-2373526943)
In all other cases we say Red Hat and not RHDH. On another note RHDH needs to be replaced by the attribute for the short name of the product.

**Review by benwilcock — APPROVED** [review]

File: modules/dynamic-plugins/ref-rh-compatible-plugins.adoc [hunk 3]
```diff
@@ -0,0 +1,40 @@
+// This page is generated! Do not edit the .adoc file, but instead run rhdh-supported-plugins.sh to regen this page from the latest plugin metadata.
+// cd /path/to/rhdh-documentation; ./modules/dynamic-plugins/rhdh-supported-plugins.sh; ./build/scripts/build.sh; google-chrome titles-generated/main/plugin-rhdh/index.html
+
+= Other compatible plugins
```
Comment (nickboldt) [inline modules/dynamic-plugins/ref-rh-compatible-plugins.adoc:4]:
```suggestion
= Other compatible installable plugins 
```

File: modules/dynamic-plugins/ref-rh-compatible-plugins.adoc [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// This page is generated! Do not edit the .adoc file, but instead run rhdh-supported-plugins.sh to regen this page from the latest plugin metadata.
+// cd /path/to/rhdh-documentation; ./modules/dynamic-plugins/rhdh-supported-plugins.sh; ./build/scripts/build.sh; google-chrome titles-generated/main/plugin-rhdh/index.html
+
+= Other compatible plugins 
+
+{product-very-short} provides Technology Preview support for the following compatible plugins:
```
Comment (nickboldt) [inline modules/dynamic-plugins/ref-rh-compatible-plugins.adoc:6]:
```suggestion
{product-very-short} provides Technology Preview support for the following compatible plugins, which are not preinstalled and must be enabled from an external source:
```

**Review by Gerry-Forde — APPROVED** [review]
/lgtm

**Comment by reestr** [comment]
Approved for the language of 2.4 - other compatible plugins, @nickboldt .

**Review by themr0c — APPROVED** [review]

**Pull request merged**